### PR TITLE
[codex] Fall back to review nodes for change-requested PRs

### PR DESCRIPTION
**Pull request by redreceipt**
## Summary

Add a fallback for active change-request detection when GitHub does not return an aggregate `reviewDecision`.

## Why

PR #319 moved the stale review notifier to GitHub's current `reviewDecision`, which correctly suppresses reminders while a PR is actively changes-requested and allows reminders again once GitHub reports `REVIEW_REQUIRED`. A late review note identified an edge case: repositories without required PR reviews can return `reviewDecision: null` even when `reviews.nodes` contains a `CHANGES_REQUESTED` review.

## Changes

- Keep `reviewDecision` as the authoritative current state when GitHub returns it.
- Fall back to fetched `reviews.nodes` only when `reviewDecision` is missing or null.
- Add a regression test for the null-decision fallback.

## Validation

- `.venv-ci/bin/python -m unittest discover -s tests -p 'test_*.py'`
- `.venv-ci/bin/ruff check .`
- `.venv-ci/bin/ruff format --check .`
- `.venv-ci/bin/vulture . --config pyproject.toml`

### CODE REVIEW

File: github.py [hunk 1]
```diff
@@ -200,7 +200,14 @@ def has_known_merge_conflicts(pr):
...
 
-    return pr.get("reviewDecision") == "CHANGES_REQUESTED"
+    review_decision = pr.get("reviewDecision")
+    if review_decision:
+        return review_decision == "CHANGES_REQUESTED"
+
+    return any(
+        review.get("state") == "CHANGES_REQUESTED"
+        for review in pr.get("reviews", {}).get("nodes", [])
+    )
```
[the remark this answers is not in the document]
Comment (redreceipt) [reply github.py]: Addressed in `8599740`: the fallback now compares `CHANGES_REQUESTED` review `submittedAt` against the latest review-request event when `reviewDecision` is missing. That keeps current change requests suppressed, but allows a re-requested reviewer to become eligible again when the only change-request review is historical. Added `test_waiting_for_review_ignores_historical_review_nodes_for_missing_decision` for that flow.

File: github.py [hunk 2]
```diff
@@ -212,6 +207,33 @@ def _parse_github_timestamp(value: str | None) -> datetime | None:
...
+        return review_decision == "CHANGES_REQUESTED"
+
+    review_request_times = [
+        requested_at
+        for review_request in pr.get("timelineItems", {}).get("nodes", [])
+        if review_request.get("requestedReviewer")
+        for requested_at in [_parse_github_timestamp(review_request.get("createdAt"))]
+        if requested_at is not None
+    ]
+    latest_review_request_at = max(review_request_times, default=None)
```
[the remark this answers is not in the document]
Comment (redreceipt) [reply github.py]: Addressed in `684c3b9`: the fallback now indexes review-request times by reviewer login and only treats a historical `CHANGES_REQUESTED` review as cleared when that same reviewer has a later review-request event. Added `test_waiting_for_review_ignores_unrelated_review_requests_for_missing_decision` so an unrelated later request cannot clear another reviewer’s active change request.
